cmd/juju/status.go

```diff
@@ -375,16 +402,18 @@ func (sf *statusFormatter) formatService(name string, service api.ServiceStatus)
 
 func (sf *statusFormatter) formatUnit(unit api.UnitStatus, serviceName string) unitStatus {
 	out := unitStatus{
-		Err:            unit.Err,
-		AgentState:     unit.AgentState,
-		AgentStateInfo: sf.getUnitStatusInfo(unit, serviceName),
-		AgentVersion:   unit.AgentVersion,
-		Life:           unit.Life,
-		Machine:        unit.Machine,
-		OpenedPorts:    unit.OpenedPorts,
-		PublicAddress:  unit.PublicAddress,
-		Charm:          unit.Charm,
-		Subordinates:   make(map[string]unitStatus),
+		Err:                unit.Err,
+		WorkloadStatusInfo: sf.getWorkloadStatusInfo(unit),
+		AgentStatusInfo:    sf.getAgentStatusInfo(unit),
+		AgentState:         unit.AgentState, // < 1.24 agent-state
+		AgentStateInfo:     sf.getUnitStatusInfo(unit, serviceName),
```

Comment:
Similar comment here - AgentStateInfo vs AgentStatusInfo etc. getUnitStatusInfo() vs getAgentStatusInfo(). Perhaps getUnitStatusInfo() could be getLegacyAgentState() or something?